library/alloc/src/slice.rs

```diff
@@ -304,7 +308,9 @@ impl<T> [T] {
     /// handled without allocation, medium sized slices allocate `self.len()` and beyond that it
     /// clamps at `self.len() / 2`.
     ///
-    /// If `K: Ord` does not implement a total order, the implementation may panic.
+    /// # Panics
+    ///
+    /// May panic if `K: Ord` does not implement a total order.
```

Comment:
This seems confusingly worded, to me.
- Ord "by definition" is a total order.
- The name of the type variable is K, but you write `K: Ord` here, suggesting that is somehow different from `K`.

When discussing something bounded by Ord, because the trait defines itself as "implemented for traits that have a total order", you are normally allowed to assume that type has a total order. This is discussing the situation where that assumption is violated. That doesn't really happen for K, because it is a parameter, not a concrete instance. It doesn't happen for Ord, because that's a trait, not its implementation. So I think it also doesn't happen for `K: Ord`, but rather happens for a "fully resolved" instance where K has been substituted with a specific type and a specific implementation of Ord.

So I think you should simply say "the implementation of Ord for K". Discussing it as an `impl Ord for K` seems also fine, since it evokes what matters: the definition site, not the bounding site.

Reply:
I like "the implementation of Ord for K", done.

Reply:
done